core/api/src/services/ledger/paginated-ledger.ts

```diff
@@ -9,34 +9,34 @@ import { parseFilterQuery } from "medici/build/helper/parse/parseFilterQuery"
 
 import { MainBook } from "./books"
 
-import { InvalidPaginationArgumentsError } from "@/domain/ledger"
+import { translateToLedgerTx } from "."
+
 import { Transaction } from "@/services/ledger/schema"
-import { MAX_PAGINATION_PAGE_SIZE } from "@/config"
+import { checkedToPaginatedQueryCursor } from "@/domain/primitives"
 
-type IFilterQuery = {
-  account?: string | string[]
-  _journal?: Types.ObjectId | string
-  start_date?: Date | string | number
-  end_date?: Date | string | number
-} & Partial<ILedgerTransaction>
+type LedgerQueryFilter = {
+  mediciFilters: {
+    account?: string | string[]
+    _journal?: Types.ObjectId | string
+    start_date?: Date | string | number
+    end_date?: Date | string | number
+  }
+  username?: string
+  addresses?: OnChainAddress[]
+}
 
 export const paginatedLedger = async ({
-  query,
+  filters,
   paginationArgs,
 }: {
-  query: IFilterQuery
-  paginationArgs?: PaginationArgs
-}): Promise<Error | PaginatedArray<ILedgerTransaction>> => {
-  const filterQuery = parseFilterQuery(query, MainBook)
-
-  const { first, after, last, before } = paginationArgs || {}
-
-  if (
-    (first !== undefined && last !== undefined) ||
-    (after !== undefined && before !== undefined)
-  ) {
-    return new InvalidPaginationArgumentsError()
-  }
+  filters: LedgerQueryFilter
+  paginationArgs: PaginatedQueryArgs
+}): Promise<Error | PaginatedQueryResult<LedgerTransaction<WalletCurrency>>> => {
+  const filterQuery = parseFilterQuery(filters.mediciFilters, MainBook)
+
+  const { first, last, before, after } = paginationArgs
+
+  filterQuery["_id"] = { $exists: true }
```

Comment:
is this needed?

Reply:
Technically the "_id" field is optional in the schema, so I added it in.